components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/connector/SelfRegistrationConfigImplTest.java

```diff
@@ -157,6 +161,10 @@ public void testGetPropertyNames() {
         propertiesExpected
                 .add(IdentityRecoveryConstants.ConnectorConfig.SELF_REGISTRATION_SMSOTP_VERIFICATION_CODE_EXPIRY_TIME);
         propertiesExpected.add(IdentityRecoveryConstants.ConnectorConfig.SELF_REGISTRATION_CALLBACK_REGEX);
+        propertiesExpected.add(LIST_PURPOSE_PROPERTY_KEY);
```

Comment:
What is this used for?
Is this necessary?

Reply:
It is necessary to add to test case  because it is already added property in the selfRegistrationConfigImpl.getPropertyNames() [line 170]